eventsverify.go

```diff
@@ -0,0 +1,186 @@
+package veracity
+
+import (
+	"context"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/datatrails/go-datatrails-merklelog/massifs"
+	"github.com/datatrails/go-datatrails-merklelog/massifs/snowflakeid"
+	"github.com/datatrails/go-datatrails-merklelog/mmr"
+	"github.com/datatrails/go-datatrails-simplehash/simplehash"
+	"github.com/urfave/cli/v2"
+)
+
+var (
+	ErrVerifyInclusionFailed = errors.New("the entry is not in the log")
+	ErrUncommittedEvents     = errors.New("one or more events did not have record of their inclusion in the log")
+)
+
+const (
+	skipUncommittedFlagName = "skip-uncommitted"
+)
+
+// NewEventsVerifyCmd verifies inclusion of a DataTrails event in the tenants Merkle Log
+//
+//nolint:gocognit
+func NewEventsVerifyCmd() *cli.Command {
+	return &cli.Command{
+		Name:    "events-verify",
+		Aliases: []string{"everify"},
+		Usage: `verify the inclusion of an event, or list of events, in the tenant's merkle log.
+
+The event response data from the DataTrails get event or list event queries can be provided directly.
+
+See the README for example use.
+
+Note: for publicly attested events, or shared protected events, you must use --tenant. Otherwise, the tenant is inferred from the event data.
+`,
+		Flags: []cli.Flag{
+			&cli.BoolFlag{Name: skipUncommittedFlagName, Value: false},
+		},
+		Action: func(cCtx *cli.Context) error {
+
+			var err error
+
+			cmd := &CmdCtx{}
+
+			// This command uses the structured logger for all optional output.
+			// Output not explicitly printed is silenced by default.
+			if err = cfgLogging(cmd, cCtx); err != nil {
+				return err
+			}
+
+			verifiableEvents, err := readArgs0FileOrStdIo(cCtx)
+			if err != nil {
+				return err
+			}
+
+			if err = cfgMassifReader(cmd, cCtx); err != nil {
+				return err
+			}
+
+			log := func(m string, args ...any) {
+				cmd.log.Infof(m, args...)
+			}
+
+			proofPath := func(proof [][]byte) string {
+				var hexProof []string
+				for _, node := range proof {
+					hexProof = append(hexProof, fmt.Sprintf("%x", node))
+				}
+				return fmt.Sprintf("[%s]", strings.Join(hexProof, ", "))
+			}
+
+			tenantIdentity := cCtx.String("tenant")
+			if tenantIdentity != "" {
+				log("verifying for tenant: %s", tenantIdentity)
+			} else {
+				log("verifying protected events for the asset creator")
+			}
+
+			var countNotCommitted int
+			var countVerifyFailed int
+
+			for _, event := range verifiableEvents {
```

Comment:
i think we should use the logverification module for this?

having 2 implementations of verifying a list of events might be more maintenance than needed?

Reply:
after a conversation this isn't doing a verification on a complete list of events, it purposefully ignores omissions.

We should do a follow up at some point to move this logic into logverification :)